### PR TITLE
Fixed Local client not sending Connect event in HostServer Mode + Added Test

### PR DESCRIPTION
**Pull request by zwazel**

### CODE REVIEW

File: lightyear/src/tests/host_server_stepper.rs [hunk 1]
```diff
@@ -247,11 +247,10 @@ impl HostServerStepper {
         self.client_app
             .world_mut()
             .run_system_once(|mut commands: Commands| commands.connect_client());
-
         // Advance the world to let the connection process complete
         for _ in 0..100 {
             if self
-                .client_app
+                .server_app
```
Comment (cBournhonesque) [inline lightyear/src/tests/host_server_stepper.rs:253]: this should be `self.client_app`, because the external client needs to check if they're synced

**Review by cBournhonesque — COMMENTED** [review]
I have one comment but other this LGTM; thanks!

File: lightyear/src/tests/host_server_stepper.rs (hunk 1 above)
Comment (zwazel) [reply lightyear/src/tests/host_server_stepper.rs]: yep, originally didnt want to push this change, just overlooked it. fixed now :)

**Comment by cBournhonesque** [comment]
Thanks, great addition :)

**Pull request merged**